compiler/circle-mpqsolver/src/Methods/BisectionSolver.cpp

```diff
@@ -0,0 +1,291 @@
+/*
+ * Copyright (c) 2022 Samsung Electronics Co., Ltd. All Rights Reserved
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "BisectionSolver.h"
+#include "../utils/ErrorApproximator.h"
+#include "../utils/ModuleCloner.h"
+#include "../utils/Evaluator.h"
+
+#include <luci/ImporterEx.h>
+#include <luci/Importer.h>
+
+#include <luci/CircleExporter.h>
+#include <luci/CircleFileExpContract.h>
+// #include <luci/CircleQuantizer.h>
+
+#include <cmath>
+
+using namespace mpqsolver;
+
+using NodeDepthType = std::map<luci::CircleNode *, float>;
+
+namespace
+{
+
+int compute_depth(const luci::Module *module, NodeDepthType &nodes_depth, float &min_depth,
+                  float &max_depth)
+{
+  if (module->size() != 1)
+    return EXIT_FAILURE;
+
+  const loco::Graph *graph = module->graph(0);
+  if (!graph)
+    return EXIT_FAILURE;
+
+  auto nodes = graph->nodes();
+  uint32_t nodes_size = nodes->size();
+  std::set<std::string> input_names;
+  {
+    auto inp_nodes = graph->inputs();
+    for (uint32_t i = 0; i < inp_nodes->size(); ++i)
+    {
+      auto inp_node = inp_nodes->at(i);
+      auto inp_name = inp_node->name();
+      input_names.insert(inp_name);
+    }
+  }
+
+  std::set<luci::CircleNode *> inputs;
+  std::map<luci::CircleNode *, std::vector<luci::CircleNode *>> inputs_to_outputs;
+  for (uint32_t node_index = 0; node_index < nodes_size; ++node_index)
+  {
+    auto node = loco::must_cast<luci::CircleNode *>(nodes->at(node_index));
```

Comment:
You can use `loco` library.
```suggestion
  for (auto node : loco::all_nodes(graph))
  {
    auto cnode = loco::must_cast<const luci::CircleNode *>(node);
```
And, `loco::input_nodes` could be used as well.